ITL-1

UltraMKF12 opened the pull request.

UltraMKF12 committed:
ITL-7 Create the header component

Also deleted default some auto-generated vue components and a files that came with the project setup.

UltraMKF12 committed:
ITL-8 Add the title

Added the title as an h1 element.

UltraMKF12 committed:
ITL-9 Add an add button, include a icon library or include the icon from the design

Stored the svg in a seperate file. Also made layout more similar to the one seen in figma.

UltraMKF12 committed:
ITL-10 Create the event that triggers the todo creation functionality

The button, when pressed, emits addTodo

zsolt-wbg commented on src/App.vue:
```diff
@@ -1,34 +1,9 @@
-<script setup lang="ts"></script>
+<script setup lang="ts">
+import TodoHeader from './components/TodoHeader.vue'
```
Please add this code block in tsconfig.json in order to use import paths with '@/...' 
`"paths": {
            "@/*": [
                "src/*"
            ]
        },`

zsolt-wbg commented on src/App.vue:
```diff
@@ -1,34 +1,9 @@
-<script setup lang="ts"></script>
+<script setup lang="ts">
```
The order in a vue document should be: <template>, <script> and <style>

zsolt-wbg commented on src/components/TodoHeader.vue:
```diff
@@ -0,0 +1,13 @@
+<script setup lang="ts">
```
Same here

zsolt-wbg commented on src/components/TodoHeader.vue:
```diff
@@ -0,0 +1,13 @@
+<template>
+  <div class="flex justify-between">
+    <h1 class="text-7xl font-bold">To do List</h1>
+    <button @click="$emit('addTodo')">
```
Create Base button component if you intend to use it more than one time.

zsolt-wbg commented on src/components/icons/plus_icon.vue:
```diff
@@ -0,0 +1,13 @@
+<template>
```
Please name vue components with PascalCase and ts files with kebab-case

zsolt-wbg commented on src/components/icons/plus_icon.vue:
```diff
@@ -0,0 +1,13 @@
+<template>
+  <svg width="60" height="60" viewBox="0 0 60 60" fill="none" xmlns="http://www.w3.org/2000/svg">
```
Please add HeroIcons to the project in order to use icons via the projects
https://heroicons.com/

UltraMKF12 committed:
Added requested changes

szabolcs3 commented on src/components/icons/PlusIcon.vue:
```diff
@@ -0,0 +1,13 @@
+<template>
```
You can delete this file.

szabolcs3 commented on src/components/TodoHeader.vue:
```diff
@@ -0,0 +1,14 @@
+<template>
+  <div class="flex justify-between">
+    <h1 class="text-7xl font-bold">To do List</h1>
+    <BaseButton>
```
You don't need to wrap around the icon with button

szabolcs3 commented on src/components/BaseButton.vue:
```diff
@@ -0,0 +1,5 @@
+<template>
+  <slot></slot>
```
Please use `<button>` html tag here and add some attributes to it so you can easily change the button attributes.
```
Example: 

- Props (label, disabled, text color, background color, button text)
- Emits (click)
 ```

UltraMKF12 committed:
Added requested changes

Deleted the icon.vue, the PlusButtonIcon is now outside of BaseButton, modified base button according to the request, but don't know if it's correct.

zsolt-wbg commented on src/components/TodoHeader.vue:
```diff
@@ -0,0 +1,11 @@
+<template>
+  <div class="flex justify-between">
+    <h1 class="text-7xl font-bold">To do List</h1>
+    <PlusCircleIcon @click="$emit('addTodo')" class="h-16 w-16 text-green-400 cursor-pointer" />
```
Please use the correct attribute order! Here is an example: https://github.com/vuejs/eslint-plugin-vue/blob/master/docs/rules/attributes-order.md
But we will send you the "Vue Bible" :)

zsolt-wbg replied:
Please use a separate function for emit, for instance:
`@click="addTodo"`
`function addTodo() {
   emit('addTodo')
}`

zsolt-wbg commented on src/components/BaseButton.vue:
```diff
@@ -0,0 +1,16 @@
+</template>
+
+<script setup lang="ts">
+interface Props {
```
Please add the props directly to the button element tag, like this:
```
<button
   :label="label"
   :disabled="disabled"
   :class="[textColor, backgroundColor]"
>
      { buttonText }
</button>
```

UltraMKF12 committed:
Added requested changes

Also made ESLint stricter, added tailwind prettier plugin that helps with reordering tailwind classes, enabled one attribute per line Prettier rule

UltraMKF12 committed:
Deleted label from BaseButton

Deleted label prop/property from base button
